### PR TITLE
FYLE-41ggca: Comments and history

### PR DESCRIPTION
**Pull request by Dimple16**

### CODE REVIEW

**Review by Julias0 — APPROVED** [review]

**Comment by Dimple16** [comment]
User comments - Android
Screenshot 2021-07-10 at 1 05 10 AM

**Comment by Dimple16** [comment]
**User comments - IOS**
Screenshot 2021-07-10 at 11 27 05 AM

File: src/app/core/services/status.service.ts [hunk 1]
```diff
@@ -44,91 +44,91 @@ export class StatusService {
       case (lowerCaseComment.indexOf('hotel request') > -1) || (lowerCaseComment.indexOf('transportation request') > -1):
         statusCategory = {
           category: 'Others',
-          icon: 'fy-info-circle-o'
+          icon: 'circle'
```
Comment (Dimple16) [inline src/app/core/services/status.service.ts:47]: this is not being used for now. will update these with new icons.

**Comment by Dimple16** [comment]
**History - Android**
Screenshot 2021-07-12 at 8 40 52 PM

**Comment by Dimple16** [comment]
**History - IOS**
Screenshot 2021-07-12 at 8 53 36 PM

File: src/app/fyle/my-reports/my-reports-card/my-reports-card.component.html [hunk 2]
```diff
@@ -44,8 +44,8 @@
   </div>
   <div class="reports-card-menu--buttons">
     <div matRipple class="reports-card-menu--action reports-card-menu--action__comments">
-      <app-comments objectType="reports" [objectId]="erpt.rp_id" mode="{{(['APPROVER_PENDING', 'APPROVER_INQUIRY'].indexOf(erpt.rp_state) === -1) ? 'view' : 'edit'}}" text="View Comments" [hideIcon]="true">
-      </app-comments>
+      <app-comments-history objectType="reports" [objectId]="erpt.rp_id" mode="{{(['APPROVER_PENDING', 'APPROVER_INQUIRY'].indexOf(erpt.rp_state) === -1) ? 'view' : 'edit'}}" text="View Comments" [hideIcon]="true">
```
Comment (Julias0) [inline src/app/fyle/my-reports/my-reports-card/my-reports-card.component.html:47]:
```
[mode]="(['APPROVER_PENDING', 'APPROVER_INQUIRY'].indexOf(erpt.rp_state) === -1) ? 'view' : 'edit'"
```

File: src/app/shared/components/comments-history/audit-history/audit-history.component.scss [hunk 3]
```diff
@@ -0,0 +1,41 @@
+@import '../../../../../theme/colors.scss';
+
+.audit-history {
+  padding: 0 22px 12px 22px;
+
+  &--container {
+    border-left: 2px solid #ECECEC;
```
Comment (Julias0) [inline src/app/shared/components/comments-history/audit-history/audit-history.component.scss:7]: move the colors into variables

File: src/app/fyle/my-reports/my-reports-card/my-reports-card.component.html (hunk 2 above)
Comment (Dimple16) [reply src/app/fyle/my-reports/my-reports-card/my-reports-card.component.html:47]: mode is not used now, missed removing from here,

File: src/app/shared/components/comments-history/audit-history/audit-history.component.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  templateUrl: './audit-history.component.html',
+  styleUrls: ['./audit-history.component.scss'],
+})
+export class AuditHistoryComponent implements OnInit {
+
+  @Input() estatuses: ExtendedStatus[];
+
+  constructor() { }
+
+  hasDetails(estatus: ExtendedStatus) {
```
Comment (Julias0) [inline src/app/shared/components/comments-history/audit-history/audit-history.component.ts:15]: not happy about how this is being used. You can create a method inside the `ExtendedStatus` class to do this.

File: src/app/shared/components/comments-history/audit-history/statuses-diff/statuses-diff.component.html [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<div *ngIf="isValueList" class="statuses-diff">
+  <div class="statuses-diff--expense-issues">
+      <span *ngIf="key === 'Please contact your admin to configure the following'">{{ key | snakeCaseToSpaceCase }} </span>
+      <span *ngIf="key != 'Please contact your admin to configure the following'" class="text-capitalize">{{ key | snakeCaseToSpaceCase }}</span>
+  </div>
+  <div *ngFor="let value1 of value">
```
Comment (Julias0) [inline src/app/shared/components/comments-history/audit-history/statuses-diff/statuses-diff.component.html:6]: rename value and value1 to be context specifc

File: src/app/shared/components/comments-history/view-comment/view-comment.component.scss [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+  }
+
+  &--self-comment {
+    background-color: #F7F7FB;
+    margin-left: 24px;
+    margin-right: 10px;
+  }
+
+  &--others-comment {
+    background-color: #FBF7F9;
```
Comment (Julias0) [inline src/app/shared/components/comments-history/view-comment/view-comment.component.scss:36]: colors move into colors.scss

File: src/app/shared/components/comments-history/audit-history/audit-history.component.ts [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  hasDetails() {
+    this.estatuses = this.estatuses.map(function (estatus) {
+      if (estatus) {
+        estatus.has_details = (estatus.st_diff !== null && Object.keys(estatus.st_diff).length > 0);
+      }
+      return estatus;
+    });
+  };
+
+  setReimbursable() {
```
Comment (Julias0) [inline src/app/shared/components/comments-history/audit-history/audit-history.component.ts:24]: setupReimbursable()

File: src/app/shared/components/comments-history/audit-history/audit-history.component.ts [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  templateUrl: './audit-history.component.html',
+  styleUrls: ['./audit-history.component.scss'],
+})
+export class AuditHistoryComponent implements OnInit {
+
+  @Input() estatuses;
+
+  constructor() { }
+
+  hasDetails() {
```
Comment (Julias0) [inline src/app/shared/components/comments-history/audit-history/audit-history.component.ts:15]: setupHasDetails()

**Review by Julias0 — APPROVED** [review]

**Pull request merged**